### PR TITLE
Export Selector Data should be combined with Discovery Data #5492

### PR DESCRIPTION
**Pull request by Aias00**
Make sure that:

- [x] You have read the [contribution guidelines](https://shenyu.apache.org/community/contributor-guide).
- [x] You submit test cases (unit or integration tests) that back your changes.
- [x] Your local test passed `./mvnw clean install -Dmaven.javadoc.skip=true`.

### CODE REVIEW

**Review by moremind — APPROVED** [review]

File: shenyu-admin/src/main/java/org/apache/shenyu/admin/controller/ConfigsExportImportController.java [hunk 1]
```diff
@@ -65,7 +71,7 @@ public ConfigsExportImportController(final ConfigsService configsService) {
     @GetMapping("/export")
     public ResponseEntity<byte[]> exportConfigs(final HttpServletResponse response) {
         ShenyuAdminResult result = configsService.configsExport();
-        if (!Objects.equals(CommonErrorCode.SUCCESSFUL, result.getCode())) {
+        if (!result.getCode().equals(CommonErrorCode.SUCCESSFUL)) {
```
Comment (moremind) [inline shenyu-admin/src/main/java/org/apache/shenyu/admin/controller/ConfigsExportImportController.java:74]: not change

File: shenyu-admin/src/main/java/org/apache/shenyu/admin/controller/ConfigsExportImportController.java [hunk 2]
```diff
@@ -97,7 +103,12 @@ public ShenyuAdminResult importConfigs(final MultipartFile file) {
             return ShenyuAdminResult.error(ShenyuResultMessage.PARAMETER_ERROR);
         }
         try {
-            return configsService.configsImport(file.getBytes());
+            ShenyuAdminResult importResult = configsService.configsImport(file.getBytes());
+            if (importResult.getCode().equals(CommonErrorCode.SUCCESSFUL)) {
+                // sync data
```
Comment (moremind) [inline shenyu-admin/src/main/java/org/apache/shenyu/admin/controller/ConfigsExportImportController.java:108]: should avoid npe

**Review by moremind — CHANGES_REQUESTED** [review]

**Comment by Aias00** [comment]
cause i thought that the import method should keep clean，just do save data logic。 and the sync event will be done after import success

---- Replied Message ----
| From | ***@***.***> |
| Date | 03/19/2024 21:41 |
| To | apache/shenyu ***@***.***> |
| Cc | aias00 ***@***.***>,
Author ***@***.***> |
| Subject | Re: [apache/shenyu] Export Selector Data should be combined with Discovery Data #5492 (PR #5493) |

@moremind commented on this pull request.

In shenyu-admin/src/main/java/org/apache/shenyu/admin/service/impl/PluginServiceImpl.java:

> @@ -280,7 +280,6 @@ public ConfigImportResult importData(final List<PluginDTO> pluginList) {
                 pluginId = pluginDO.getId();
                 if (pluginMapper.insertSelective(pluginDO) > 0) {
                     // publish create event. init plugin data
-                    pluginEventPublisher.onCreated(pluginDO);


why remove?

**Review by moremind — APPROVED** [review]

**Comment by ywj1352** [comment]
@Aias00  Can we close it now？

**Comment by Aias00** [comment]
> @Aias00 Can we close it now？

yeap, already done

**Pull request merged**